tests/test_notify_slack.py

```diff
@@ -113,6 +131,54 @@ def should_not_notify_slack_when_step_function_running(webhook_client_mock: Magi
         webhook_client_mock.assert_not_called()
 
 
+@patch("backend.notify_status_update.task.WebhookClient.send")
+@patch("backend.notify_status_update.task.get_import_status_given_arn")
+def should_not_notify_slack_when_step_function_failed(
+    step_func_status_mock: MagicMock, webhook_client_mock: MagicMock
+) -> None:
+    # Given
+
+    webhook_client_mock.return_value.status_code = HTTPStatus.OK
+    mock_slack_url = any_https_url()
+    now_ts = datetime.now()
+
+    step_func_status_mock.return_value = {
+        STEP_FUNCTION_KEY: {STATUS_KEY: JOB_STATUS_FAILED},
+        VALIDATION_KEY: {STATUS_KEY: Outcome.SKIPPED.value, ERRORS_KEY: []},
+        METADATA_UPLOAD_KEY: {STATUS_KEY: Outcome.SKIPPED.value, ERRORS_KEY: []},
+        ASSET_UPLOAD_KEY: {STATUS_KEY: Outcome.SKIPPED.value, ERRORS_KEY: []},
+    }
+
+    with patch.dict(environ, {SLACK_URL_ENV_NAME: mock_slack_url}), patch(
+        "backend.notify_status_update.task.publish_sns_message"
+    ):
+        # When
+        notify_status_update_input = {
+            EVENT_DETAIL_KEY: {
+                STATUS_KEY: JOB_STATUS_FAILED,
+                STEP_FUNCTION_ARN_KEY: any_arn_formatted_string(),
+                STEP_FUNCTION_INPUT_KEY: dumps(
+                    {
+                        DATASET_ID_KEY: any_dataset_id(),
+                        DATASET_PREFIX_KEY: any_dataset_prefix(),
+                        VERSION_ID_KEY: any_dataset_version_id(),
+                    }
+                ),
+                STEP_FUNCTION_STARTDATE_KEY: round(
+                    (now_ts - timedelta(seconds=10)).timestamp() * 1000
+                ),
+                STEP_FUNCTION_STOPDATE_KEY: round(now_ts.timestamp() * 1000),
+            },
+            STEP_FUNCTION_OUTPUT_KEY: None,
+        }
+
+        lambda_handler(notify_status_update_input, any_lambda_context())
+
+        # Then
+        webhook_client_mock.assert_called_once()
+        assert len(webhook_client_mock.call_args[1][WEBHOOK_MESSAGE_BLOCKS_KEY]) == 9
```

Comment:
I don't understand this assertion (and the corresponding one in `should_notify_slack_with_finished_details_when_url_set`). Would it be easier to understand if we asserted the actual values of (a subset of) the arguments? If there's no easy way to make this readable to someone unfamiliar with `WebhookClient.send`, could you please add some context to the "Then" comment above?

Reply:
Hey, yeah It's asserting the number of message 'blocks' (concept from slack_sdk) sent in the request. I decided to do this cause it would potentially be overkill to assert the blocks individually. I could assert some text values from the blocks, for example that the first block text is "Geostore Dataset Version Import" etc. What do you think?

Reply:
Ah, right. Nah, it's a bit clearer now.